src/ClientData/CaptureData.cpp

```diff
@@ -76,31 +77,21 @@ void CaptureData::ForEachThreadStateSliceIntersectingTimeRange(
 }
 
 const ScopeStats& CaptureData::GetScopeStatsOrDefault(ScopeId scope_id) const {
-  static const ScopeStats kDefaultScopeStats;
-  auto scope_stats_it = scope_stats_.find(scope_id);
-  if (scope_stats_it == scope_stats_.end()) {
-    return kDefaultScopeStats;
-  }
-  return scope_stats_it->second;
+  return all_scopes_->GetScopeStatsOrDefault(scope_id);
 }
 
 void CaptureData::UpdateScopeStats(const TimerInfo& timer_info) {
   const std::optional<ScopeId> scope_id = ProvideScopeId(timer_info);
   if (!scope_id.has_value()) return;
 
-  ScopeStats& stats = scope_stats_[scope_id.value()];
-  const uint64_t elapsed_nanos = timer_info.end() - timer_info.start();
-  stats.UpdateStats(elapsed_nanos);
+  all_scopes_->UpdateScopeStats(scope_id.value(), timer_info);
 }
 
 void CaptureData::AddScopeStats(ScopeId scope_id, ScopeStats stats) {
-  scope_stats_.insert_or_assign(scope_id, stats);
+  all_scopes_->SetScopeStats(scope_id, stats);
 }
 
-void CaptureData::OnCaptureComplete() {
-  thread_track_data_provider_->OnCaptureComplete();
-  UpdateTimerDurations();
```

Comment:
You probably need to still call to SortTimerDurations here, unless you want to move this post-process to a user-action. But in this case, please explain why you want that in the PR body.

Reply:
Done